### PR TITLE
[Config] Extra tests for Config component

### PR DESCRIPTION
**Pull request by zomberg**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |

### CODE REVIEW

File: src/Symfony/Component/Config/Tests/Definition/Builder/ExprBuilderTest.php [hunk 1]
```diff
@@ -151,6 +151,30 @@ public function testThenUnsetExpression()
...
+
+    /**
+     * @expectedException \RuntimeException
+     * @expectedExceptionMessage You must specify a then part.
+     */
+    public function testEndThenPartNotSpecified()
+    {
+        $builder = $this->getTestBuilder();
+
+        $reflection = new \ReflectionProperty($builder, 'ifPart');
```
Comment (unkind) [inline src/Symfony/Component/Config/Tests/Definition/Builder/ExprBuilderTest.php]: Why do you need reflection here?

Comment (zomberg) [inline src/Symfony/Component/Config/Tests/Definition/Builder/ExprBuilderTest.php]: Thanks. You are right. I didn't notice that this is public property.

**Comment by javiereguiluz** [comment]
👍 

Status: reviewed

@zomberg thanks for adding these missing tests!

File: src/Symfony/Component/Config/Tests/Definition/Builder/ArrayNodeDefinitionTest.php [hunk 2]
```diff
@@ -185,6 +185,49 @@ public function testTrueEnableEnabledNode($expected, $config, $message)
...
+    public function testCanBeDisabled()
+    {
+        $node = new ArrayNodeDefinition('root');
+        $node->canBeDisabled();
+
+        $this->assertTrue($this->getField($node, 'addDefaults'));
+        $this->assertEquals(array('enabled' => false), $this->getField($node, 'falseEquivalent'));
+        $this->assertEquals(array('enabled' => true), $this->getField($node, 'trueEquivalent'));
+        $this->assertEquals(array('enabled' => true), $this->getField($node, 'nullEquivalent'));
+        $this->assertEquals(array('enabled' => true), $this->getField($node, 'nullEquivalent'));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Config/Tests/Definition/Builder/ArrayNodeDefinitionTest.php]: duplicated line?

**Comment by nicolas-grekas** [comment]
:+1: with minor comment

File: src/Symfony/Component/Config/Tests/Definition/Builder/ArrayNodeDefinitionTest.php (hunk 2 above)
Comment (zomberg) [inline src/Symfony/Component/Config/Tests/Definition/Builder/ArrayNodeDefinitionTest.php]: @nicolas-grekas, done.

**Comment by nicolas-grekas** [comment]
Thank you @zomberg.

**Pull request closed without merging**